install.sh

```diff
@@ -61,178 +66,33 @@ for dir in "${CREW_CONFIG_PATH}/meta" "${CREW_DEST_DIR}" "${CREW_PACKAGES_PATH}"
-  sha256s+=('982db8c305a1c2859ab9954a18e03e5500a277fa00717e7646bf69b95a744213')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/libunistring-0.9.10-chromeos-x86_64.tar.xz')
-  sha256s+=('d0fccb9b680e38517ce95483fcbac289ba835aa79e5c106bc21ec8de7872a13b')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/openldap-2.4.57-chromeos-x86_64.tar.xz')
-  sha256s+=('e0ca5de88bd9808f8e30beb0861bec2da9c3686b538f4563603b19fbd50b6278')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/rtmpdump-c5f04a58f-1-chromeos-x86_64.tar.xz')
-  sha256s+=('0212408c6faad92b5909d77188d8c27e68ce955f61dcb5597603303e31e601ce')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/zstd-1.4.7-chromeos-x86_64.tar.xz')
-  sha256s+=('017e2d9e5fa613bb50385f6b49bb9642f7b3e6f10453e46b89ea3e4701a39ae7')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/ncurses-6.2-20210206-1-chromeos-x86_64.tar.xz')
-  sha256s+=('d81416072e4e4454b585495db8b7e0767867c851e512afd4655fa8174d244c02')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/ca_certificates-20210119-chromeos-x86_64.tar.xz')
-  sha256s+=('7a782fdba9f89b077004dab99c15367bb999ed1b19485e32a22be2e79c4f3ac3')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/ruby-3.0.0-2-chromeos-x86_64.tar.xz')
-  sha256s+=('20d4d18d1a184a088abb5547bc94e7f6189229565157bb35c5691cf0f1264f9f')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/libffi-3.3-chromeos-x86_64.tar.xz')
-  sha256s+=('aecc094eb0e5b961106ee3d4086f120a30555cb82b3b8263107d9cab4e6c278f')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/openssl-1.1.1i-1-chromeos-x86_64.tar.xz')
-  sha256s+=('c5d19a3cab5f38898f4875ca2be57c3a8fb23c73726dd14e29f7bed02d252eb0')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/nettle-3.7-1-chromeos-x86_64.tar.xz')
-  sha256s+=('713dbccd1a267bac1cd71856ab6f68dd897ae32a005d2c9439f9d31d31597481')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/curl-7.75.0-chromeos-x86_64.tar.xz')
-  sha256s+=('a511db565f40b830a623a173739d29a3c935ea2698ebb407b65a630d1e279b17')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/git-2.30.0-2-chromeos-x86_64.tar.xz')
-  sha256s+=('3a63be93d3bbc46811a7f8b81a8e4b08c03ff82facf36419030ac214b60e4e6b')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/icu4c-68.2-chromeos-x86_64.tar.xz')
-  sha256s+=('290087e9c03a94eaf5d33a82b76dc3b5bb11be1dff5a6e57aa190ec5439bcfb2')
-  urls+=('https://dl.bintray.com/chromebrew/chromebrew/libedit-20191231-3.1-chromeos-x86_64.tar.xz')
-  sha256s+=('834915020f3fd98b0e3ca8609ae0b915b47ca77ea5877fa7b758a85571401e8e')
   ;;
 esac
 
+for package in $EARLY_PACKAGES; do
+  pkgfile="$(curl -Lsf "${URL}"/packages/"$package".rb)"
+  temp_url="$(echo "$pkgfile" | grep -m 3 "$ARCH": | head -n 1 | awk '{print $2}' | tr -d \' | sed 's/,//g')"
+  temp_sha256="$(echo "$pkgfile" | grep -m 3 "$ARCH": | tail -n 1 | awk '{print $2}' | tr -d \' | sed 's/,//g')"
```

Comment:
Simplify?
```
  temp_url="$(grep -m 3 "$ARCH": "$pkgfile" | head -n 1 | cut -d\' -f2 | tr -d \' | sed 's/,//g')"
  temp_sha256="$(grep -m 3 "$ARCH": "$pkgfile" | tail -n 1 | cut -d\' -f2 | tr -d \' | sed 's/,//g')"
```
This makes the assumption there are no extra lines in ALL the binaries.  Since this has been automated for the most part, I think it is safe.  Great idea!

Reply:
ok that doesn't actually work, because "$pkgfile" is not the file name but the contents of the package file.
```
chronos@cheekon-i686 /usr/local/lib/crew/packages $ export pkgfile="$(cat ca_certificates.rb)"
chronos@cheekon-i686 /usr/local/lib/crew/packages $ echo "$pkgfile" | grep -m 3 "$ARCH": | head -n 1 | awk '{print $2}' | tr -d \' | sed 's/,//g'
https://salsa.debian.org/debian/ca-certificates
chronos@cheekon-i686 /usr/local/lib/crew/packages $ grep -m 3 "$ARCH": "$pkgfile" | head -n 1 | cut -d\' -f2 | tr -d \' | sed 's/,//g'
grep: require 'package'

class Ca_certificates < Package
  description 'CA Certificates'
  homepage 'https://salsa.debian.org/debian/ca-certificates'
  @_ver = '20210119'
  version @_ver
  compatibility 'all'
  source_url "https://salsa.debian.org/debian/ca-certificates/-/archive/debian/#{@_ver}/ca-certificates-debian-#{@_ver}.tar.bz2"
  source_sha256 'af30b4d9a2c58e42134067d29f0ba6120e5960fd140393d5574d4bdcf5b824d6'

  binary_url ({
     aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/ca_certificates-20210119-chromeos-armv7l.tar.xz',
      armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/ca_certificates-20210119-chromeos-armv7l.tar.xz',
        i686: 'https://dl.bintray.com/chromebrew/chromebrew/ca_certificates-20210119-chromeos-i686.tar.xz',
      x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/ca_certificates-20210119-chromeos-x86_64.tar.xz',
  })
  binary_sha256 ({
     aarch64: '75d332ca6053e5e12d34eb5d72da6daab50d650f6733d1bd9668bed9af802e30',
      armv7l: '75d332ca6053e5e12d34eb5d72da6daab50d650f6733d1bd9668bed9af802e30',
        i686: 'e61ae07f85f64f0c53fa922714ac30579c77db5f2e9fa266e4e6059b502f5475',
      x86_64: '7a782fdba9f89b077004dab99c15367bb999ed1b19485e32a22be2e79c4f3ac3',
  })

  def self.patch
    system "sed -i 's,/usr/share/ca-certificates,#{CREW_PREFIX}/share/ca-certificates,g' \
      Makefile"
    system "sed -i 's,/usr/share/ca-certificates,#{CREW_PREFIX}/share/ca-certificates,g' \
      sbin/update-ca-certificates"
    system "sed -i 's,CERTSCONF=/etc/ca-certificates.conf,CERTSCONF=#{CREW_PREFIX}/etc/ca-certificates.conf,g' \
      sbin/update-ca-certificates"
    system "sed -i 's,ETCCERTSDIR=/etc/ssl/certs,ETCCERTSDIR=#{CREW_PREFIX}/etc/ssl/certs,g' \
      sbin/update-ca-certificates"
    system "sed -i 's,HOOKSDIR=/etc/ca-certificates/update.d,HOOKSDIR=#{CREW_PREFIX}/etc/ca-certificates/update.d,g' \
      sbin/update-ca-certificates"
    system "sed -i '/restorecon/d' sbin/update-ca-certificates"
    system "sed -i 's,/usr/sbin,#{CREW_PREFIX}/bin,g' sbin/Makefile"
  end

  def self.build
    system 'make'
  end

  def self.install
    FileUtils.mkdir_p("#{CREW_DEST_PREFIX}/etc/ssl/certs/")
    FileUtils.mkdir_p("#{CREW_DEST_PREFIX}/bin")
    FileUtils.mkdir_p("#{CREW_DEST_PREFIX}/share/ca-certificates/")
    system "touch #{CREW_DEST_PREFIX}/etc/ca-certificates.conf"
    system "make DESTDIR=#{CREW_DEST_DIR} install"
    FileUtils.ln_s Dir.glob("#{CREW_PREFIX}/share/ca-certificates/mozilla/*"),
                   "#{CREW_DEST_PREFIX}/etc/ssl/certs/"
  end

  def self.postinstall
    system 'update-ca-certificates --hooksdir "" --fresh'
  end
end: No such file or directory
```

But I did put in the awk replacement, and added another tr filter to remove double quotes.